mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala

```diff
@@ -493,6 +493,25 @@ abstract class PcCollector[T](
         case df: NamedDefTree
             if df.span.isCorrect && df.nameSpan.isCorrect &&
               filter(df) && !isGeneratedGiven(df) =>
+          def collectEndMarker =
+            val name = df.name.toString()
+            val endMarker = s"end $name"
```

Comment:
Could this potentially break if we have more spaces between `end` and `name` ?

Reply:
Or a trailing whitespace at the end of the line? :slightly_smiling_face:

Reply:
Maybe we should use regex instead?

Reply:
I would avoid createc new regexes hre, we can just split of \s+ and check the two resulting parts

Reply:
In theory there can also be a comment there. Regex would take care of that too. Are you worried about the performance? I think it should be fine, since it would be a simple regex. But I can parse it by hand too.